### PR TITLE
refactor: standardize naming

### LINKED ISSUE #186: Standardize path, collectionPath, and filename
There seems to be very little convention around paths, collectionPaths and filenames all accross the code, we should standardize. My suggestions: 

- Use `path` as the full path inside a collection **including the filename** (e.g. `/dir1/dir2/file.txt`)
- Use `filename` as a variable name instead of `file`, `name` and `filename` scattered around the codebase

The above would allow us to treat files and directories much more easily with packages like [`path`](https://pkg.go.dev/path)

### PR DESCRIPTION
**Pull request by aeryz**
Namings in the API are standardized. Following endpoints are effected:

1. `POST /content/add-ipfs` -> `name` is changed to `filename`
2. `POST /content/add` -> `name` is changed to `filename`.
3. `GET /content/stats` -> `file` is changed to `filename` in response.
4. `GET /collections/fs/list` -> `col` is changed to `collection` and `dir` is changed to `collectionPath`.
5. `GET /collections/fs/add` -> `col` (which is the shorthand for `collection`) is removed. 

Fixes #186 and #173

Note that path params like `coluuid` are not changed since they are not directly used by users. I can change those as well though.

Signed-off-by: aeryz <abdullaheryz@protonmail.com>

### CODE REVIEW

**Review by gmelodie — APPROVED** [review]
LGTM!

File: docs/docs.go [hunk 1]
```diff
@@ -209,14 +219,14 @@ const docTemplate_swagger = `{
                     {
                         "type": "string",
                         "description": "Collection",
-                        "name": "col",
```
Comment (gmelodie) [inline docs/docs.go:212]: `collection` or `coluuid`?

File: docs/swagger.json [hunk 2]
```diff
@@ -65,6 +65,23 @@
...
+        "description": "This endpoint is used to delete an existing collection.",
+        "tags": [
+          "collections"
+        ],
+        "summary": "Deletes a collection",
+        "parameters": [
+          {
+            "type": "string",
+            "description": "Collection ID",
+            "name": "coluuid",
```
Comment (gmelodie) [inline docs/swagger.json:79]: `collection` or `coluuid`?

File: docs/swagger.json [hunk 3]
```diff
@@ -202,14 +212,14 @@
           {
             "type": "string",
             "description": "Collection",
-            "name": "col",
+            "name": "collection",
```
Comment (gmelodie) [inline docs/swagger.json:215]: `collection` or `coluuid`?

File: docs/swagger.yaml [hunk 4]
```diff
@@ -118,6 +118,18 @@ paths:
       tags:
         - admin
   /collections/add-content:
+    delete:
+      description: This endpoint is used to delete an existing collection.
+      parameters:
+        - description: Collection ID
+          in: path
+          name: coluuid
```
Comment (gmelodie) [inline docs/swagger.yaml:126]: `collection` or `coluuid`?

File: docs/swagger.yaml [hunk 5]
```diff
@@ -228,12 +235,12 @@ paths:
       parameters:
         - description: Collection
           in: query
-          name: col
+          name: collection
```
Comment (gmelodie) [inline docs/swagger.yaml:238]: `collection` or `coluuid`?

File: docs/swagger.yaml [hunk 6]
```diff
@@ -228,12 +235,12 @@ paths:
       parameters:
         - description: Collection
```
Comment (gmelodie) [inline docs/swagger.yaml:236]: This should say `Collection ID`

File: docs/swagger.json [hunk 7]
```diff
@@ -202,14 +212,14 @@
           {
             "type": "string",
             "description": "Collection",
```
Comment (gmelodie) [inline docs/swagger.json:214]: This should say `Collection ID`

File: docs/docs.go [hunk 8]
```diff
@@ -209,14 +219,14 @@ const docTemplate_swagger = `{
                     {
                         "type": "string",
                         "description": "Collection",
```
Comment (gmelodie) [inline docs/docs.go:221]: This should say `Collection ID`

File: cmd/barge/client.go [hunk 9]
```diff
@@ -298,7 +298,7 @@ type collectionListResponse struct {
 
 func (c *EstClient) CollectionsListDir(ctx context.Context, col, path string) ([]collectionListResponse, error) {
 	var out []collectionListResponse
-	_, err := c.doRequest(ctx, "GET", fmt.Sprintf("/collections/fs/list?col=%s&dir=%s", col, url.PathEscape(path)), nil, &out)
+	_, err := c.doRequest(ctx, "GET", fmt.Sprintf("/collections/fs/list?collection=%s&collectionPath=%s", col, url.PathEscape(path)), nil, &out)
```
Comment (gmelodie) [inline cmd/barge/client.go:301]: `collection` or `coluuid`?

File: docs/swagger.yaml [hunk 10]
```diff
@@ -197,11 +209,6 @@ paths:
       description: This endpoint adds a file to a collection
       parameters:
         - description: Collection ID
-          in: query
-          name: col
-          required: true
-          type: string
-        - description: Collection ID Long
           in: query
           name: collection
```
Comment (gmelodie) [inline docs/swagger.yaml:213]: `collection` or `coluuid`?

File: handlers.go [hunk 11]
```diff
@@ -4860,18 +4860,13 @@ func sanitizePath(p string) (string, error) {
...
 // @Param        path query string true "Path to file"
 // @Produce      json
 // @Router       /collections/fs/add [post]
 func (s *Server) handleColfsAdd(c echo.Context, u *User) error {
-	colid := c.QueryParam("col")
-	colidlong := c.QueryParam("collection")
-	if colid == "" {
-		colid = colidlong
-	}
+	colid := c.QueryParam("collection")
```
Comment (gmelodie) [inline handlers.go:4869]: `coluuid` here

**Review by gmelodie — CHANGES_REQUESTED** [review]
I did a more careful review. I think we should use `coluuid` instead of `collection` for the UUID of the collection, and adjust the docs to always say "Collection ID" instead of just "Collection". `collection` and "Collection" make me think of the collection object, not the UUID of the collection.

**Comment by aeryz** [comment]
> I did a more careful review. I think we should use `coluuid` instead of `collection` for the UUID of the collection, and adjust the docs to always say "Collection ID" instead of just "Collection". `collection` and "Collection" make me think of the collection object, not the UUID of the collection.

Thanks for the in-depth review. Working on it asap.

**Comment by aeryz** [comment]
`collection` is changed to `coluuid` and docs are fixed.

We changed `collection` to `coluuid` but we use `collectionPath` for path in the collection. We might wanna change it to `colpath` to be more consistent. (We shouldn't make it just `path` because `ContentInCollection` struct is used in several other types. Changing it to `path` would create confusions.)

**Comment by gmelodie** [comment]
@aeryz `colpath` sounds like a good variable name. Go for it!

**Pull request merged**